Release 1.9 blog post (#924)
flowforge/website: src/blog/2023/07/flowforge-1-9-release.md

---
title: FlowForge now offers API Documentation with Swagger UI
subtitle: FlowForge 1.9 adds new features to make it easier to administer FlowForge
description: FlowForge 1.9 
date: 2023-07-06 
authors: ["ian-skerrett"]
image: 
tags:
    - posts
    - flowforge
    - releases
---

FlowForge 1.9 adds new features to make it easier to adminster FlowForge platform deployments, including new API documentation and the ability to create customized Node-RED palettes.

<!--more-->

> ## API Documentation

Reviewer: Once we've published the release, the docs at https://flowforge.com/docs/api/ will include more information including links to the API spec hosted of FF Cloud.

Author: To make sure I understand, the API spec for FF Cloud will allow people to create scripts against their FF Cloud accounts?

---
title: FlowForge now offers API Documentation with Swagger UI
subtitle: FlowForge 1.9 adds new features to make it easier to administer FlowForge
description: The new release of FlowForge 1.9 adds new API Swagger documentation and the ability to customize Node-RED pallettes.
date: 2023-07-06 
authors: ["ian-skerrett"]
image: 
tags:
    - posts
    - flowforge
    - releases
---

FlowForge 1.9 adds new features to make it easier to adminster FlowForge platform deployments, including new API documentation and the ability to create customized Node-RED palettes.

<!--more-->

## API Documentation

> FlowForge API allows developers to progammatically interact with the FlowForge platform. This makes it possible to integrate FlowForge into different infrastrcutre technologies, create scripts to automate specifc FlowForge tasks and embedd FlowForge into other applications.  

Reviewer 2: ```suggestion
FlowForge API allows developers to programmatically interact with the FlowForge platform. This makes it possible to integrate FlowForge into different infrastructure technologies, create scripts to automate specific FlowForge tasks and embed FlowForge into other applications.
```

In the 1.9 release we are now publishing our [API documentation](https://flowforge.com/docs/api/) using the [OpenAPI specification](https://swagger.io/specification/) and making it viewable with the Swagger UI. Both these industrial standards will make using the FlowForge API easier to use and understand.

## Customize Node-RED Palettes  [#2002](https://github.com/flowforge/flowforge/issues/2002)

> FlowForge platform adminstrators are now able to create customized Node-RED palettes that will be used when a Node-RED instance is created. An adminstator can create pre-defined templates to specify the nodes that should be included in the palette. This makes it easier for FlowForge teams to standardized on Node-RED usage across an organization.

Reviewer 3: It's not for teams, so the last sentence isn't correct? This is only useful on-prem?

Reviewer 2: See my suggestion above

> Note: this feature is not available for FlowForge cloud users since they do not have adminstrator access. In the future, the ability to create templates will be enabled for teams and this will allow for customize Node-RED palettes for individual teams.

Reviewer 2: ```suggestion
Note: this feature is not available for FlowForge cloud users since they do not have administrator access.
```

> Starting with the 1.9 release, FlowForge Cloud will begin to include the [Node-RED debugger](https://flows.nodered.org/node/node-red-debugger) and [lint node](https://github.com/node-red/nrlint) in their default templates. 

Reviewer 2: ```suggestion
Starting with the 1.9 release, FlowForge Cloud will begin to include the [Node-RED debugger](https://flows.nodered.org/node/node-red-debugger) and [Node-RED Flow Linter](https://github.com/node-red/nrlint) in their default templates. 
```

Reviewer 3: How does this work with pipelines? Will these be copied to snapshots and to production instances? Debugger not linter should be there by default?

@MarianRaphael will this break deploys for customers that use a module cache?

Reviewer 2: ```suggestion
```

Reviewer 2: Clarified via Slack. Will not going to happen in 1.9

## New RBAC Role for Dashboard users [#2292](https://github.com/flowforge/flowforge/issues/1924)
A new FlowForge user role has been created to view Node-RED dashboards. This allows for users to view Nod-RED dashboards without access to the Node-RED editor or requiring seperate login credentials.

## Other New Features

- FlowForge device agent is now supported on Windows [#78](https://github.com/flowforge/flowforge-device-agent/issues/78)
- Allow local configuration of https/httpStatic on a device [#110](https://github.com/flowforge/flowforge-device-agent/issues/110)
- Instance settings: Add https.key and https.certs [#2257](https://github.com/flowforge/flowforge/issues/2257)

> - Allow devices to access details of the snapshot running on them [#94](https://github.com/flowforge/flowforge-device-agent/issues/94)

Reviewer 2: ```suggestion
- Allow devices to access the "Snapshot ID" and the "Snapshot Name" running on them [#94](https://github.com/flowforge/flowforge-device-agent/issues/94)
```

> 

Reviewer 2: ```suggestion
- High Availability logging enhanced: Individual Node-RED Instance replica querying and filtering [https://github.com/flowforge/flowforge/issues/2260](2260)
```

In the 1.9 release we are now publishing our [API documentation](https://flowforge.com/docs/api/) using the [OpenAPI specification](https://swagger.io/specification/) and making it viewable with the Swagger UI. Both these industrial standards will make using the FlowForge API easier to use and understand.

## Customize Node-RED Palettes  [#2002](https://github.com/flowforge/flowforge/issues/2002)

FlowForge platform adminstrators are now able to create customized Node-RED palettes that will be used when a Node-RED instance is created. An adminstator can create pre-defined templates to specify the nodes that should be included in the palette. This makes it easier for FlowForge teams to standardized on Node-RED usage across an organization.

Note: this feature is not available for FlowForge cloud users since they do not have administrator access.

## New RBAC Role for Dashboard users [#2292](https://github.com/flowforge/flowforge/issues/1924)

> A new FlowForge user role has been created to view Node-RED dashboards. This allows for users to view Nod-RED dashboards without access to the Node-RED editor or requiring seperate login credentials.

Reviewer 2: ```suggestion
A new FlowForge user role has been created to view Node-RED dashboards. This allows for users to view Nod-RED dashboards without access to the Node-RED editor or requiring separate login credentials.
```
